clc/modules/block-storage/src/main/java/com/eucalyptus/blockstorage/async/SnapshotDeleter.java

```diff
@@ -170,16 +172,23 @@ private void deleteFromEBS() {
 
   private void deleteFromOSG() {
     try {
-      SnapshotInfo searchSnap = new SnapshotInfo();
-      searchSnap.setStatus(StorageProperties.Status.deletedfromebs.toString());
+      // Get the snapshots that are deleted from EBS but not yet deleted from OSG, 
+      // in reverse time order so we never try to delete a parent before a child
       List<SnapshotInfo> snapshotsToBeDeleted = null;
-      try {
-        snapshotsToBeDeleted = Transactions.findAll(searchSnap);
+      try (TransactionResource tr = Entities.transactionFor(SnapshotInfo.class)) {
+        snapshotsToBeDeleted = Entities.criteriaQuery(
+            Entities.restriction(SnapshotInfo.class)
+            .like(SnapshotInfo_.status, StorageProperties.Status.deletedfromebs.toString()).build())
```

Comment:
You'd want to use "equal" here rather than "like" as the status will not have wildcards (I don't think this causes any functionality issues in this case)